src/app/dashboard/components/filters/form.tsx

```diff
@@ -1,47 +1,108 @@
 import { zodResolver } from '@hookform/resolvers/zod/dist/zod';
-import { Button } from '@mui/material';
-import React from 'react';
+import { Dayjs } from 'dayjs';
 import { useForm, useWatch } from 'react-hook-form';
 import styled from 'styled-components';
+import { NumberParam, StringParam, useQueryParams } from 'use-query-params';
 
 import Form from 'components/form';
 import DateInput from 'components/form/dateInput';
+import NumberInput from 'components/form/numberInput';
 
 import { FiltersForm, FiltersFormSchema } from './schema';
-import { getFilterFormDefaultValues, minDateBoundary, today } from './utils';
+import { minDateBoundary, today } from './utils';
+import { useDebounce } from 'use-debounce';
+import { useEffect } from 'react';
+
+interface Props {
+  setStartDateHandle: (childData: Dayjs) => void;
+  setEndDateHandle: (childData: Dayjs) => void;
+  setAggregateHandle: (aggregate: number) => void;
+  startDate: Dayjs;
+  endDate: Dayjs;
+  aggregate: number;
+}
+
+const StockChartFiltersForm = (props: Props) => {
+  const [query, setQuery] = useQueryParams({
+    date_from: StringParam,
+    date_to: StringParam,
+    aggregate: NumberParam,
+  });
+  query;
 
-const StockChartFiltersForm: React.FC = () => {
   const methods = useForm<FiltersForm>({
-    defaultValues: getFilterFormDefaultValues(),
+    defaultValues: {
+      startDate: props.startDate,
+      endDate: props.endDate,
+      aggregate: props.aggregate,
+    },
     resolver: zodResolver(FiltersFormSchema),
   });
-  const { handleSubmit, control } = methods;
+  const { handleSubmit, control, setValue } = methods;
```

Comment:
2. Nie potrzebujemy `setValue` - wartości formularza automatycznie się aktualizują z wykorzystaniem `react-hook-form`
3. Nie potrzebujemy funkcji `handleSubmit`, bo nie mamy żadnego submit button'a więc też do StyledForm'a nie przekazujemy onSubmit, co za tym idzie trzeba poprawić type Props'ów - `onSubmit?: FormProps['onSubmit'];` - zmiana na opcjonalny
4. Jeżeli nie mamy submit buttona to tracimy walidację, więc wyciągniemy sobie pomocną funkcję z `rhf` - `trigger`, zaraz jej użyjemy

 https://react-hook-form.com/api/useform/trigger/

```suggestion
  const { control, trigger } = methods;
```

Reply:
fixed